tools/ci_build/github/linux/run_dockerbuild.sh

```diff
@@ -83,32 +81,32 @@ if [ $BUILD_OS = "yocto" ]; then
         --dockerfile $DOCKER_FILE --context .
 elif [ $BUILD_DEVICE = "gpu" ]; then
-        # This code path is only for training. Inferecing pipeline uses CentOS
-        IMAGE="$BUILD_OS-gpu_training"
-        # Current build script doesn't support building shared lib with Python dependency. To enable building with PythonOp,
-        # We need to avoid `--no-undefined` when building shared lib (Otherwise, CIs will report `undefined symbols`), but removing that would bring some other concerns.
-        # Plus the fact training did not need build shared library, we disable the --build_shared_lib for training CIs.
-        NEED_BUILD_SHARED_LIB=false
-        INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -t"
-        if [[ $INSTALL_DEPS_DISTRIBUTED_SETUP = true ]]; then
-            INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -m"
-        fi
-        if [[ $ORTMODULE_BUILD = true ]]; then
-            INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -u"
-        fi
-        INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -v 11.8"
-        $GET_DOCKER_IMAGE_CMD --repository "onnxruntime-$IMAGE" \
-            --docker-build-args="--build-arg BASEIMAGE=nvcr.io/nvidia/cuda:11.8.0-cudnn8-devel-${BUILD_OS} --build-arg BUILD_USER=onnxruntimedev --build-arg BUILD_UID=$(id -u) --build-arg PYTHON_VERSION=${PYTHON_VER} --build-arg INSTALL_DEPS_EXTRA_ARGS=\"${INSTALL_DEPS_EXTRA_ARGS}\" --build-arg USE_CONDA=${USE_CONDA} --network=host" \
-            --dockerfile Dockerfile.ubuntu_gpu_training --context .
+    # This code path is only for training. Inferecing pipeline uses CentOS
+    IMAGE="$BUILD_OS-gpu_training"
+    # Current build script doesn't support building shared lib with Python dependency. To enable building with PythonOp,
+    # We need to avoid `--no-undefined` when building shared lib (Otherwise, CIs will report `undefined symbols`), but removing that would bring some other concerns.
+    # Plus the fact training did not need build shared library, we disable the --build_shared_lib for training CIs.
+    NEED_BUILD_SHARED_LIB=false
+    INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -t"
+    if [[ $INSTALL_DEPS_DISTRIBUTED_SETUP = true ]]; then
+        INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -m"
+    fi
+    if [[ $ORTMODULE_BUILD = true ]]; then
+        INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -u"
+    fi
+    INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -v 12.1"
+    $GET_DOCKER_IMAGE_CMD --repository "onnxruntime-$IMAGE" \
+        --docker-build-args="--build-arg BASEIMAGE=nvcr.io/nvidia/cuda:12.1.0-cudnn8-devel-${BUILD_OS} --build-arg BUILD_USER=onnxruntimedev --build-arg BUILD_UID=$(id -u) --build-arg PYTHON_VERSION=${PYTHON_VER} --build-arg INSTALL_DEPS_EXTRA_ARGS=\"${INSTALL_DEPS_EXTRA_ARGS}\" --build-arg USE_CONDA=${USE_CONDA} --network=host" \
```

Comment:
upgrade to cudnn 9 image?

Reply:
there is not nvcr.io/nvidia/cuda:12.1.0-cudnn9-devel in nvcr.io

Reply:
How about nvidia/cuda:12.3.2-cudnn9-devel

Reply:
training test need install torch, while torch doesn't publish wheel for 12.3 for now.

Reply:
This is the tricky part: you won't be able to find the combination of CUDA/CUDNN version we use from official Nvidia images.